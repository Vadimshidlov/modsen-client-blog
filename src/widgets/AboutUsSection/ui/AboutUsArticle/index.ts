export { AboutUsArticle } from '@/widgets/AboutUsSection/ui/AboutUsArticle/AboutUsArticle';
