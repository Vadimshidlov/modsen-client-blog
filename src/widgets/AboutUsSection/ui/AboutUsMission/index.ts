export { AboutUsMission } from '@/widgets/AboutUsSection/ui/AboutUsMission/AboutUsMission';
