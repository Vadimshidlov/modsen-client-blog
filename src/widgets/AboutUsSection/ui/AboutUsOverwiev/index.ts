export { AboutUsOverwiev } from '@/widgets/AboutUsSection/ui/AboutUsOverwiev/AboutUsOverwiev';
