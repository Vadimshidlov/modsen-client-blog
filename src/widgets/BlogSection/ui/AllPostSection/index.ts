export { AllPostSection } from '@/widgets/BlogSection/ui/AllPostSection/AllPostSection';
