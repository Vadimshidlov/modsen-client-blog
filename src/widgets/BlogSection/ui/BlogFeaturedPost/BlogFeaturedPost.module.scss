.blogFeaturedPost {
  background: $lightPurple;

  .postBodyWrapper {
    padding: 80px;
    display: flex;
    justify-content: space-between;
    column-gap: 71px;

    @media (max-width: 1000px) {
      flex-direction: column-reverse;
      row-gap: 20px;
      padding: 60px;
    }

    @media (max-width: $tablet) {
      padding: 40px;
    }

    @media (max-width: $mobile) {
      padding: 20px;
    }

    .postContent {
      display: flex;
      flex-direction: column;

      .postTitle {
        max-width: 624px;
        @include headingTwo;
        margin-bottom: 16px;

        @media (max-width: $tablet) {
          @include headingThree;
        }

        @media (max-width: $mobile) {
          @include headingFour;
        }
      }

      .postSubTitle {
        font-size: $fz16;
        font-weight: $bold;
        line-height: 20px;
        margin-bottom: 20px;
        letter-spacing: 3px;

        @media (max-width: 768px) {
          margin-bottom: 16px;
        }

        @media (max-width: 450px) {
          margin-bottom: 10px;
        }
      }

      .postAuthor {
        @include labelOne;
        margin-bottom: 16px;

        span {
          color: $lightFiolet;
        }
      }

      .postText {
        display: block;
        max-width: 624px;
        margin-bottom: 32px;
        @include bodyOne;
      }
    }

    .postImageWrapper {
      max-width: 515px;

      @media (max-width: 1000px) {
        align-self: center;
      }

      .postImage {
        @media (max-width: $tablet) {
          height: 300px;
        }

        @media (max-width: $mobile) {
          height: 250px;
        }

        width: 100%;
        height: 359px;
        object-fit: cover;
      }
    }
  }
}
