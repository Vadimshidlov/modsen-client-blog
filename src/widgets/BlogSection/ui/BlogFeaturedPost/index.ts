export { BlogFeaturedPost } from '@/widgets/BlogSection/ui/BlogFeaturedPost/BlogFeaturedPost';
