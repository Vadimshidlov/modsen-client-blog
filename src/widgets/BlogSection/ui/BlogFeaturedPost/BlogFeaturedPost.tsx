import React from 'react';
import Image from 'next/image';
import Link from 'next/link';
import { useTranslations } from 'next-intl';
import { Inter } from 'next/font/google';
import { POSTS } from '@/shared/constants/posts';
import { Button } from '@/shared/ui/Button';
import container from '@/shared/styles/container.module.scss';
import styles from '@/widgets/BlogSection/ui/BlogFeaturedPost/BlogFeaturedPost.module.scss';

const inter = Inter({
  subsets: ['latin'],
  weight: '500',
  variable: '--font-inter',
});

export function BlogFeaturedPost() {
  const t = useTranslations('BlogFeaturedPost');
  const featuredPost = POSTS.filter((postData) => postData.id === 1)[0];
  const { author, postDate, text, title, id, img } = featuredPost;

  return (
    <section className={styles.blogFeaturedPost}>
      <div className={`${container.container} ${styles.postBodyWrapper}`}>
        <article className={styles.postContent}>
          <h3 className={`${styles.postSubTitle} ${inter.variable}`}>{t('subTitle')}</h3>
          <h2 className={`${styles.postTitle} ${inter.variable}`}>{title}</h2>
          <span className={styles.postAuthor}>
            {t('authorTag')}&nbsp;
            <span>{author}</span> |{postDate}
          </span>
          <span className={styles.postText}>{text}</span>
          <Link href={`/posts/${id}`}>
            <Button>{t('buttonText')} &gt;</Button>
          </Link>
        </article>
        <div className={styles.postImageWrapper}>
          <Image className={styles.postImage} alt="post-image" src={img} placeholder="blur" />
        </div>
      </div>
    </section>
  );
}
