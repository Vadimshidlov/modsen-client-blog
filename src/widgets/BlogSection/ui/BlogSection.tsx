import React from 'react';
import { BlogFeaturedPost } from '@/widgets/BlogSection/ui/BlogFeaturedPost';
import { AllPostSection } from '@/widgets/BlogSection/ui/AllPostSection';

export function BlogSection() {
  return (
    <section>
      <BlogFeaturedPost />
      <AllPostSection />
    </section>
  );
}
