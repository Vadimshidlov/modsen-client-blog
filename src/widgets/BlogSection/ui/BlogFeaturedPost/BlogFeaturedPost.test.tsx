import React from 'react';
import { render } from '@testing-library/react';
import { BlogFeaturedPost } from '@/widgets/BlogSection/ui/BlogFeaturedPost';

jest.mock('next-intl', () => ({
  useTranslations: jest.fn().mockReturnValue((key: string) => `Mocked ${key}`),
}));

describe('BlogFeaturedPost tests', () => {
  test('should matches snapshot', () => {
    const { container } = render(<BlogFeaturedPost />);
    expect(container).toMatchSnapshot();
  });
});
