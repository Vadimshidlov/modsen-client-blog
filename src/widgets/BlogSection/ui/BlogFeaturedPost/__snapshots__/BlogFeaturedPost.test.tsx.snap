// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogFeaturedPost tests should matches snapshot 1`] = `
<div>
  <section
    class="blogFeaturedPost"
  >
    <div
      class="container postBodyWrapper"
    >
      <article
        class="postContent"
      >
        <h3
          class="postSubTitle variable"
        >
          Mocked subTitle
        </h3>
        <h2
          class="postTitle variable"
        >
          Step-by-step guide to choosing great font pairs
        </h2>
        <span
          class="postAuthor"
        >
          Mocked authorTag
          Â 
          <span>
            Floyd Miles
          </span>
           |
          May 23, 2024
        </span>
        <span
          class="postText"
        >
          Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident.
        </span>
        <a
          href="/posts/1"
        >
          <button
            class="button"
            type="button"
          >
            Mocked buttonText
             &gt;
          </button>
        </a>
      </article>
      <div
        class="postImageWrapper"
      >
        <img
          alt="post-image"
          class="postImage"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
          width="40"
        />
      </div>
    </div>
  </section>
</div>
`;
